userapi/api/api.go

```diff
@@ -226,6 +228,27 @@ type PerformAccountDeactivationResponse struct {
 	AccountDeactivated bool
 }
 
+// PerformOpenIDTokenCreationRequest is the request for PerformOpenIDTokenCreation
+type PerformOpenIDTokenCreationRequest struct {
+	UserID string
+}
+
+// PerformOpenIDTokenCreationResponse is the response for PerformOpenIDTokenCreation
+type PerformOpenIDTokenCreationResponse struct {
+	Token OpenIDToken
+}
+
+// QueryOpenIDTokenRequest is the request for QueryOpenIDToken
+type QueryOpenIDTokenRequest struct {
+	Token string
+}
+
+// QueryOpenIDTokenResponse is the response for QueryOpenIDToken
+type QueryOpenIDTokenResponse struct {
+	Sub       string // The Matrix User ID that generated the token
+	ExpiresTS int64
```

Comment:
When using ints to describe timestamps, please put units in the field name e.g `ExpiresAtMS` would probably do the trick. `Expires` on its own is ambiguous if its a timestamp or absolute duration, but `ExpiresAt` indicates a timestamp reasonably well, so we don't have to do things like `ExpiresTSMS` :D

Reply:
That's much nicer with MS. I changed all the expiry names to match this convention.